src/Foundatio.Repositories.Elasticsearch/Configuration/DailyIndex.cs

```diff
@@ -131,9 +135,12 @@ protected override DateTime GetIndexDate(string index)
         return DateTime.MaxValue;
     }
 
-    private readonly Dictionary<DateTime, object> _ensuredDates = new();
     protected async Task EnsureDateIndexAsync(DateTime utcDate)
     {
+        await using var indexLock = await _ensureIndexLock.AcquireAsync($"Index:{GetVersionedIndex(utcDate.Date)}", TimeSpan.FromMinutes(1)).AnyContext();
```

Comment:
not in love that some date logic and this index name gets created after this, but felt like this needed to happen like this for the lock